### ISSUE TITLE
Continuous batching is loud by default: WARNING logs and progress bars on a plain generate_batch

### ISSUE THREAD

**Issue by qgallouedec** [huggingface/transformers#48313]
A plain `generate_batch` call at default verbosity prints a stream of warnings and a progress bar; `model.generate()` prints nothing. Users embedding CB in a serving loop or trainer get this on every manager start.

## Reproduction

```python
import torch
from transformers import AutoModelForCausalLM, AutoTokenizer, GenerationConfig

model_id = "Qwen/Qwen3-0.6B"
tokenizer = AutoTokenizer.from_pretrained(model_id)
model = AutoModelForCausalLM.from_pretrained(model_id, dtype=torch.bfloat16, device_map="cuda:0")

prompt = tokenizer.apply_chat_template([{"role": "user", "content": "Hi!"}], add_generation_prompt=True)["input_ids"]
model.generate_batch([prompt] * 4, generation_config=GenerationConfig(max_new_tokens=16))
```

## Outcome

```
Solving 4 requests:   0%|          | 0/4 [00:00<?, ?request/s]
ContinuousBatchingLogger - WARNING - `eos_token_id` not set in GenerationConfig. Setting to -1 (disabled).
ContinuousBatchingLogger - WARNING - Continuous batching is much better when using flash attention. Switching from sdpa to flash_attention_3. If you need to use eager or sdpa, use paged|eager or paged|sdpa as the `attn_implementation`.
[transformers] You do not have `flash_attn` installed, using `paged|kernels-community/vllm-flash-attn3` from the `kernels` library instead!
ContinuousBatchingLogger - WARNING - Warming up for continuous batching...
ContinuousBatchingLogger - WARNING - Warming up completed in 8.39s.
Solving 4 requests: 100%|██████████| 4/4 [00:10<00:00,  2.56s/request]
```

(On TP runs there is additionally `Setting NCCL_GRAPH_MIXING_SUPPORT = 0 ...` per rank, and each subsequent manager in the same process repeats the attention-switch and warmup messages.)

Most of these are informational (attention implementation switched, warmup started/finished, NCCL env tweak) and none of them describes a problem with the user's call; at default verbosity they belong at `info`/`debug`. The progress bar should follow the same convention as the rest of the library (opt-in or tied to logging verbosity), since `generate_batch` is used programmatically inside trainers and servers.

## Expected behavior

Silence at default verbosity, matching `model.generate()`. Keep genuinely actionable warnings (e.g. `eos_token_id` unset) at `warning`; move status messages to `info`; gate the `Solving n requests` tqdm behind an argument or the logging level.

## System info

transformers main (a353632), torch 2.13.0+cu130, Python 3.12.12, NVIDIA H100 80GB.
